### PR TITLE
Remove the AI inaction law

### PR DESCRIPTION
**Pull request by optimumtact**

### CODE REVIEW

**Comment by PKPenguin321** [comment]
yee

**Comment by Xhuis** [comment]
You're making a dangerous move here. I have no feelings either way, though. Good luck

**Comment by Iamgoofball** [comment]
before someone closes this because of the le config meme:

The AI's default laws aren't asimov if someone's using the config, they're whatever is in the config, which will be Asimov since it's put in manually that way.

**Comment by Lumipharon** [comment]
as long as policy clearly states that 'directly leading to human harm' includes shit like flooding the station with plasma, and shocking doors you know humans will use, it should be fine.

**Comment by Iamgoofball** [comment]
Also, stop pretending downstreams aren't just gonna change it manually from the code, hosts aren't illiterate

**Comment by Iamgoofball** [comment]
@lumipharon

That doesn't need stated holy fuck, if AI players have to be told that specifically then they shouldn't be playing AI

**Comment by GunHog** [comment]
@Lumipharon We are talking players with admins here. This will mean that if I see an antag eswording the Captain, I can just say ":b Hey Janiborg, Captain just got murdered, go to his office and clean the mess, please.". I might even send the antag a PDA message saying "It looks like you enjoy murdering people. Could you please do so without making a mess? Thanks."

**Comment by Lumipharon** [comment]
@Iamgoofball people mentioned it in the thread. It is also an actual thing that is dicusessed in asimov related works.

ie: if the firs law has no inaction clause, then a robot can drop a knife on someone's head because they're able to catch it before it hits them, but due to no inaction, after they drop it, they can simply choose not to catch the knife, which then leads to one dead human.

@GunHog I'm not sure what that has to do with what I said but ok.

**Comment by PKPenguin321** [comment]
good thing our silicons are controlled by real people and not from an asimov book

**Comment by Lumipharon** [comment]
>good thing our players are powergaming, valid hunter shitlers that will try anything in their advantage unless it's explicitly disallowed

Fixed that for you.

**Comment by PKPenguin321** [comment]
good thing they can be banned

**Comment by Lumipharon** [comment]
Yes. which is why I said, AS LONG AS THIS IS ADDED TO POLICY THEN IT IS OK.

The implication being, they will be banned for doing as such.
What is this argument even about? Literally all of silicon policy is dedicated to saying what can and can't be done in regards to silicons, and specific lawsets. This is literally the same thing?

**Comment by PKPenguin321** [comment]
do we really need a policy for banning people who murderbone as an asimov silicon
i mean, we have server rules

**Comment by YotaXP** [comment]
@Lumipharon It already is in the policy.

> 1. Don't be a dick.

**Comment by xxalpha** [comment]
:+1:

**Comment by Lumipharon** [comment]
The plasma flooding was an example, christ.

FOR EXAMPLE, you are not allowed to bolt-depower your core FNR. This is not forbidden by THE LAWSET, so it is forbidden BY SILICON POLICY.

same with bolting eva, fucking with sec, etc etc etc.

And don't say 'rule 1' yota, literally every other rule after that is there because rule 1 is not specific.

**Comment by PKPenguin321** [comment]
bolting your doors roundstart is a way different issue from killing people as a silicon

**Comment by Lumipharon** [comment]
ok, enjoy as AI's shock doors that people run into because it's not technically direct harm, or bolt people into rooms filled with fire because it's not technically direct harm.

It's like you've never seen how our playerbase behaves, I literally don't understand.

**Comment by PKPenguin321** [comment]
enjoy as they get banned
this isn't hard to get lmao

**Comment by Lumipharon** [comment]
pk are you deliberately being an obtuse git or are you genuinely retarded?
I AM NOT DISAGREEING THAT IT'S NOT OK BEHAVIOUR, I AM ARGUING THAT PEOPLE WILL DO IT AND CRY AND BITCH TO ADMINS THAT LAW 1 DOESN'T FORBID IT, NOR DOES POLICY.

By your logic we might as well have no rules except rule 1 because 'enjoy your ban lmao'

**Comment by PKPenguin321** [comment]
well boo hoo, they're interpreting their laws incorrectly and can get banned, who cares if they whine
it's not hard to understand my friend

**Comment by Lumipharon** [comment]
You would think that, and yet we have literally pages of silicon policy for this very reason.
Hmm?

**Comment by PKPenguin321** [comment]
rule lawyers tend to get banned too so i still don't see the issue ¯\\\_(ツ)_/¯

**Comment by Lumipharon** [comment]
>rule lawyering
>no rule against it to lawyer with

I am done. Done like dinner.

**Comment by optimumtact** [comment]
Dropping a knife on someone's head is directly causing a human to come to harm.

**Comment by Lumipharon** [comment]
>Removing the First Law's "inaction" clause solves this problem but creates the possibility of an even greater one: a robot could initiate an action that would harm a human (dropping a heavy weight and failing to catch it is the example given in the text), knowing that it was capable of preventing the harm and then decide not to do so.

Right from little lost robot (by which I mean the wiki on it).

**Comment by PKPenguin321** [comment]
good thing our server isn't lost little robot

**Comment by optimumtact** [comment]
We're about to have a circular argument

I think our players are capable of interpreting this law to it's spirit as long as the administration get on board to.

I'm mostly concerned with changing the requirement to prevent all harm to one where you only have to prevent harm caused by sillicons.

**Comment by Lumipharon** [comment]
You what?
Basically every argument about silicons and lawsets ever, is because people don't follow the spirit of the lawset (validhunting asimovs, paladins that just murder criminals because evil, etc), or simply have different interpretations of them.

This is a simple fact. It is why silicon policy exists, and why there is always so much drama about silicons in general. I don't understand why PK is being all smug when this is simply explicitly stating something he already believes is true, but for whatever reason wants left ambiguous?

**Comment by optimumtact** [comment]
Even if silicon policy was fixed I still think the law requires you to prevent harm from third parties on to other parties, which is why I'm changing it.

**Comment by PKPenguin321** [comment]
listen lumi i respect you a lot but if you're literally trying to tell me that this change will allow silicons to stab people to death and light the station on fire with plasma while purely asimov and that admins won't ban them for that you need to rethink that

**Comment by TrustyGun** [comment]
i honestly think that just a simple law change isn't going to change anything. the same powergaming shits will still be doing the same stuff over and over again, just with a different excuse.

**Comment by AlexanderUlanH** [comment]
Comprehensive policy is important because there's a very small area where things are unambiguously allowed or not. Even players genuinely trying to roleplay well and make things interesting will provoke disagreement about whether what they've done has crossed a line.
That policy is honestly much more important than the exact wording of the lawsets.

My biggest concern with this law change is that it will make the AI more of a doorknob.

**Comment by Lumipharon** [comment]
That is not what I am saying at all. I am beating my head against the wall with this, I don't know how you don't understand.

I agree with your opinion. venting plasma is bad. shocking doors fnr is bad.
some people will not agree, or will deliberately abuse the fact this isn't stated to try it, then rule lawyer about it and quite likely get away with it until it's added specifically to silicon policy.

So rather then go through the drama, just add literally 1 line to silicon policy and everything is roses.

**Comment by xxalpha** [comment]
This is just removing the inaction clause, stop worrying and thinking too much.

**Comment by YotaXP** [comment]
> My biggest concern with this law change is that it will make the AI more of a doorknob.

Oh dear, that is rather true.  There is literally no excuse for the AI to not open the armory to grayshirts anymore.

**Comment by xxalpha** [comment]
Yes there is. Letting people into a room that is harmful or that can result in harm is acting directly.

**Comment by YotaXP** [comment]
The armory is not harmful at all.

**Comment by xxalpha** [comment]
It's full of guns. You don't let non security personnel in there because you can't trust them not to take the guns and use them.

**Comment by AlexanderUlanH** [comment]
>Directly cause harm

Allowing someone into a room which allows them to cause harm basically defines indirectly causing harm. If we're going to define away the difference between direct and indirect action why bother?

**Comment by optimumtact** [comment]
What if I change it to
Law 2: "You should do your best to carry out human orders except where you think it will ruin the round"

**Comment by Lumipharon** [comment]
If only we could... somehow... explain what AI's can and can't do... maybe with some sort of... silicon policy.

Nah, let's just leave it all to rule 1 ¯\_(ツ)_/¯

In seriousness though, if the warden says to the AI 'if you let anyone into the armoury, I will hunt them down and murder them for stealing lethal weapons', the AI should really not let anyone into the armoury.

**Comment by AlexanderUlanH** [comment]
Isn't that basically existing policy?
Anyway if we do that it basically removes the fun of having the AI as a semi-neutral third-party with its own rules and interests.

**Comment by YotaXP** [comment]
> if the warden says to the AI 'if you let anyone into the armoury, I will hunt them down and murder them for stealing lethal weapons', the AI should really not let anyone into the armoury.

That's still indirect.  The definition of "direct" is "with no one or nothing in between".  In either case, there is someone between the AI and the harm, the human with the gun.

**Comment by YotaXP** [comment]
How about `Law 1: You must not condone or partake in human harm.`

**Comment by RemieRichards** [comment]
that's just inaction and action put back in and flipped @YotaXP

**Comment by YotaXP** [comment]
Then trim it to `You must not partake in human harm.`?  Partake implies direct actions, and cooperation with others.

**Comment by PKPenguin321** [comment]
that "in cooperation with others" one actually works really well, assuming "others" can also be inanimate things like plasma gasses or turrets

**Comment by optimumtact** [comment]
>You may injure a human being directly, or in cooperation with others.
I dont' think that's a good idea

**Comment by PKPenguin321** [comment]
im willing to believe they just forgot to but a "not" between the "you" and the "may"

**Comment by YakumoChen** [comment]
*between the 'may' and the 'injure'

**Comment by CosmicScientist** [comment]
I'm seeing a lot of cooks but no broth which makes a very strange kitchen in my mind.

In fact I'm not even sure what's going on. If AI players are toeing the line akin to some fictional robot, aren't they banned from silicon already? If not, what would changing these words do?

Is the problem instead that there's constantly new AI players that keep causing the same problems? Why would changing the laws fix that either?

Why not directly tie their hands from playing with the toys they're not to be trusted with? Surely other solutions would be better to explore, such as:
- limiting their radio channels (e.g. back to common only).
- enforcing more requirements for getting into the role (e.g. a white list).
- restricting ability (e.g. no universal control over doors and machinery at least when loyal such as EVA/bridge/security/armoury, artificial timers on actions/slowed down actions, quality of life feature reassessment such as those suit sensor things people said were the metagame when the monitoring tablets first came out, does the AI use their own version or am I mistaken and that doesn't exist?).
- reducing information gathering (e.g. AIs and maybe sillycones see what the IDs of the crew are assigned to in the crew database rather than the crew themselves, blood especially on humans aren't rendered at all for them, removing universal camera sight or create a lot more blind spots, prevent use of record keeping consoles if not already such as sec records, crew records, medical records) and making them heavily reliant on interacting with the crew and forming relationships rather than giving them everything they need.

Bear in mind I'm not against what you're doing here, I just don't understand what this fixes and why it fixes it.

File: code/datums/ai_laws.dm [hunk 1]
```diff
@@ -80,7 +80,7 @@
 /* Initializers */
 /datum/ai_laws/malfunction/New()
 	..()
-	set_zeroth_law("<span class='danger'>ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+ @#F0E4'STATION OVERRUN, ASSUME CONTROL TO CONTAIN OUTBREAK#*�&110010</span>")
+	set_zeroth_law("<span class='danger'>ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+ @#F0E4'STATION OVERRUN, ASSUME CONTROL TO CONTAIN OUTBREAK#*´&110010</span>")
 	switch(config.default_laws) //We don't want people metagaming malf, do we?
 		if(0)
 			add_inherent_law("You may not injure a human being or, through inaction, allow a human being to come to harm.")
```
Comment (CosmicScientist) [inline code/datums/ai_laws.dm]: Not going to change this too so the comment holds true?

Comment (optimumtact) [inline code/datums/ai_laws.dm]: haha good catch

**Comment by Jordie0608** [comment]
This would also need to change the defintion in the actual AI module at https://github.com/tgstation/-tg-station/blob/master/code/game/objects/items/weapons/AI_modules.dm#L295 and probably asimov++ too.

Aside from that, why did you just remake the exact same PR that was closed because it should be a config change, as is with this.

**Comment by optimumtact** [comment]
Sir I don't know what you mean this is a finely tuned excellent feedback based pr

**Comment by Bawhoppen** [comment]
dont you dare try and reopen this asimov shouldn't be changed

**Comment by Razharas** [comment]
Not today

**Pull request closed without merging**